client/src/visualiser-src/graph-visualiser/data-structure/Edge.ts

```diff
@@ -0,0 +1,26 @@
+import { Path, SVG } from '@svgdotjs/svg.js';
+
+export class Edge {
+  public source: string;
+
+  public target: string;
+
+  public weight: number;
+
+  public isBidirectional: boolean;
+
+  private constructor(source: string, target: string, weight: number, isBidirectional: boolean) {
+    this.source = source;
+    this.target = target;
+    this.weight = weight;
+    this.isBidirectional = isBidirectional;
+  }
+
+  public static from(source: number, target: number, weight: number, isBidirectional: boolean) {
+    return new Edge(String(source), String(target), weight, isBidirectional);
+  }
+
+  public static getDomReference(source: number, target: number): Path {
```

Comment:
This is a bit hard to do since I sadly have a cyclic dependency where `renderForceGraph` requires `GraphicalEdge` to render the graph, but `GraphicalEdge` requires the graph to be rendered before the svg.js reference can be stored.

I've made it an instance method instead which dynamically queries for the SVG element, which is okay since the graph would have rendered by the time the reference is needed. 

Let me know if you have ideas (could cache the SVG reference if doing the query each time is dragging down performance).